docs/docsite/rst/os_guide/windows_performance.rst

```diff
@@ -20,24 +20,50 @@ seconds.
 
 .. code-block:: powershell
 
-    function Optimize-PowershellAssemblies {
-      # NGEN powershell assembly, improves startup time of powershell by 10x
-      $old_path = $env:path
-      try {
-        $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
-        [AppDomain]::CurrentDomain.GetAssemblies() | % {
-          if (! $_.location) {continue}
-          $Name = Split-Path $_.location -leaf
-          if ($Name.startswith("Microsoft.PowerShell.")) {
-            Write-Progress -Activity "Native Image Installation" -Status "$name"
-            ngen install $_.location | % {"`t$_"}
-          }
-        }
-      } finally {
-        $env:path = $old_path
-      }
+    function Optimize-Assemblies {
+        param (
+            [string]$assemblyFilter = "Microsoft.PowerShell.",
+            [string]$activity = "Native Image Installation"
+        )
+        
+        # Save the current $env:path
+        $old_path = $env:path
+        
+        try {
+            # Set $env:path to the runtime directory
+            $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
```

Comment:
I know this was carried over from the old script, but instead of setting the `PATH` and restoring it later, why don't we invoke `ngen` at this particular path and avoiding mutating the environment at all?

Reply:
@briantist 
You're absolutely correct. I've updated my submission to account for this.